### PR TITLE
UIOR-277 change field

### PR DESCRIPTION
**Pull request by Feboll**
## Purpose

When creating/editing an order, replace the vendor lookup with a filterable select list and ensure that the select list only shows active vendor organizations, and NOT non-vendor organizations.
https://issues.folio.org/browse/UIOR-277

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/PurchaseOrder/util.js [hunk 1]
```diff
@@ -70,3 +70,11 @@ export const getAddresses = (addresses) => {
     };
   });
 };
+
+export const getVendors = (vendors) => {
+  return vendors.filter(vendor => !vendor.isVendor && vendor.status === 'Active')
+    .map(vendor => ({
+      value: vendor.id,
+      label: vendor.name,
```
Comment (aliaksei-chumakou) [inline src/components/PurchaseOrder/util.js]: I think label should be more like this https://github.com/folio-org/ui-invoice/blob/master/src/common/utils/getOptions.js#L3

File: src/components/PurchaseOrder/util.js [hunk 2]
```diff
@@ -70,3 +70,11 @@ export const getAddresses = (addresses) => {
     };
   });
 };
+
+export const getVendors = (vendors) => {
+  return vendors.filter(vendor => !vendor.isVendor && vendor.status === 'Active')
```
Comment (aliaksei-chumakou) [inline src/components/PurchaseOrder/util.js]: Please check the condition of filtering. I think you need only vendors here, or I'm wrong? 
https://github.com/folio-org/ui-invoice/blob/master/src/invoices/InvoiceForm/InvoiceForm.js#L134

File: test/bigtest/tests/create-order-test.js [hunk 3]
```diff
@@ -68,17 +68,17 @@ describe('Create order', function () {
...
-      await form.createOrderButton.click();
-    });
-
-    it('displays list of orders, new order is created ', () => {
-      expect(orders.$root).to.exist;
-      expect(form.isPresent).to.be.false;
-      expect(orders.orders().length).to.be.equal(1);
-    });
-  });
+  // describe('Create new order', () => {
```
Comment (aliaksei-chumakou) [inline test/bigtest/tests/create-order-test.js]: please leave a comment with `TODO: to fix test with selecting required vendor field`

File: src/components/PurchaseOrder/util.js (hunk 2 above)
Comment (Feboll) [reply src/components/PurchaseOrder/util.js]: I need to shows active vendor organizations, and NOT non-vendor organizations.

File: src/common/POFields/FieldVendor.js [hunk 4]
```diff
@@ -3,18 +3,27 @@ import PropTypes from 'prop-types';
...
 import FieldSelection from '../FieldSelection';
 
-const FieldVendor = ({ vendors }) => {
+const FieldVendor = ({ vendors, name, required, validate }) => {
   return (
     <FieldSelection
       dataOptions={vendors}
       labelId="ui-orders.orderDetails.vendor"
-      name="vendorName"
+      name={name}
```
Comment (NikitaSedyx) [inline src/common/POFields/FieldVendor.js]: please keep the same name for both forms

File: src/common/POFields/FieldVendor.js [hunk 5]
```diff
@@ -3,18 +3,27 @@ import PropTypes from 'prop-types';
...
+      required={required}
```
Comment (NikitaSedyx) [inline src/common/POFields/FieldVendor.js]: I'm not sure we should validate vendor on templates

Comment (Feboll) [reply src/common/POFields/FieldVendor.js]: When need create/edit order you must have vendor

Comment (aliaksei-chumakou) [reply src/common/POFields/FieldVendor.js]: let's keep it as required for now

File: src/components/PurchaseOrder/PODetails/PODetailsForm.js [hunk 6]
```diff
@@ -216,20 +162,12 @@ class PODetailsForm extends Component {
...
-              component={TextField}
-              disabled
-              endControl={!isOpenedOrder && this.vendorClearButton()}
-              fullWidth
-              hasClearIcon={false}
-              label={<FormattedMessage id="ui-orders.orderDetails.vendor" />}
-              name="vendorName"
+            <FieldVendor
+              vendors={vendorOptions}
+              name="vendor"
```
Comment (NikitaSedyx) [inline src/components/PurchaseOrder/PODetails/PODetailsForm.js]: not required now

File: src/components/PurchaseOrder/PODetails/PODetailsForm.js [hunk 7]
```diff
@@ -216,20 +162,12 @@ class PODetailsForm extends Component {
...
               required
```
Comment (NikitaSedyx) [inline src/components/PurchaseOrder/PODetails/PODetailsForm.js]: seems the same for other props

File: src/components/PurchaseOrder/POForm.js [hunk 8]
```diff
@@ -167,6 +168,7 @@ class POForm extends Component {
     const prefixesSetting = getSettingsList(get(parentResources, 'prefixesSetting.records', {}));
     const suffixesSetting = getSettingsList(get(parentResources, 'suffixesSetting.records', {}));
     const addresses = getAddresses(get(parentResources, 'addresses.records', []));
+    const vendors = get(parentResources, 'vendors.records', []).filter(vendor => vendor.isVendor && vendor.status === ORGANIZATION_STATUS_ACTIVE);
```
Comment (NikitaSedyx) [inline src/components/PurchaseOrder/POForm.js]: break the line please

**Comment by aliaksei-chumakou** [comment]
@Feboll Could you please be more descriptive in PR? 
Now its _UIOR-277 change field_
_UIOR-277 improve PO vendor lookup field_ sounds better, IMHO.

**Pull request merged**